### DISCUSSION TITLE
GitHub-Hosted Runner issue?

### DISCUSSION THREAD

**Discussion by tamer73** [shivammathur/setup-php#822] [Q&A / Help]
hi there,

first of all thanks for sharing your work.

This is my first workflow with github and i appreciate any help.

I'm getting following error with:
| Laravel with `MySQL` and `Redis`       | `ubuntu`                        | [laravel-mysql.yml](./examples/laravel-mysql.yml "GitHub Action for Laravel with MySQL and Redis")  

on GitHub-Hosted Runner.

Did i missed sth or is this an error?
```
#Run php -r "file_exists('.env') || copy('.env.example', '.env');"
  php -r "file_exists('.env') || copy('.env.example', '.env');"
  shell: /usr/bin/bash -e {0}
  env:
    COMPOSER_PROCESS_TIMEOUT: 0
    COMPOSER_NO_INTERACTION: [1](https://github.com/feelgood-it/jhb/actions/runs/8044294546/job/21967663457#step:4:1)
    COMPOSER_NO_AUDIT: 1
PHP Warning:  copy(.env.example): Failed to open stream: No such file or directory in Command line code on line 1
[2](https://github.com/feelgood-it/jhb/actions/runs/8044294546/job/21967663457#step:4:2)s
Run composer install -q --no-ansi --no-interaction --no-scripts --no-progress --prefer-dist
  composer install -q --no-ansi --no-interaction --no-scripts --no-progress --prefer-dist
  shell: /usr/bin/bash -e {0}
  env:
    COMPOSER_PROCESS_TIMEOUT: 0
    COMPOSER_NO_INTERACTION: 1
    COMPOSER_NO_AUDIT: 1
  
Error: Process completed with exit code [1](https://github.com/feelgood-it/jhb/actions/runs/8044294546/job/21967663457#step:5:1).
```

**Comment by shivammathur** [comment]
Please check that you have a checkout step and your project has a .env.example file.

**Reply by tamer73** [reply]
thank you -  im one step further now :-) Even when i dont know what you mean with checkout step.

my issues now are:
```
> @php artisan package:discover --ansi
Could not open input file: artisan
Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 1
Error: Process completed with exit code 1.
```

where can i change this?
`Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/cache@v3. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.`

**Comment by tamer73** [comment]
thank you -  im one step further now :-)

`composer could not find a composer.json file ` but i have put some in the workflow folder beneath working .env.example.

where can i change this?
`Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/cache@v3. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.`
